File: src/relative_abundance/analysis/plot_sample_count.r
# Load required packages
library(ggplot2)
library(dplyr)

sample_csv = "./root:Host-associated:Human:Digestive%20system:Large%20intestine:Fecal_first.csv"

# Read the CSV file into a data frame
studies <- read.csv("./root:Host-associated:Human:Digestive%20system:Large%20intestine:Fecal_first.csv", stringsAsFactors = FALSE)

# Rename columns for easier use
studies <- studies %>%
  rename(
    study_name = attributes.study.name,
    samples_count = attributes.samples.count
  )

# Convert samples_count to numeric (if not already)
studies$samples_count <- as.numeric(studies$samples_count)

# Plot a horizontal bar graph of all studies
ggplot(studies, aes(x = reorder(study_name, samples_count), y = samples_count)) +
  geom_bar(stat = "identity", fill = "steelblue") +
  coord_flip() +  # horizontal bars for better readability
  labs(title = "Samples Count for 320 Studies",
       x = "Study Name",
       y = "Samples Count") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 6))  # adjust text size as needed


# Filter and sort to get the top 20 studies
top_studies <- studies %>%
  arrange(desc(samples_count)) %>%
  head(20)

# Plot the top 20 studies as a horizontal bar chart
ggplot(top_studies, aes(x = reorder(study_name, samples_count), y = samples_count)) +
  geom_bar(stat = "identity", fill = "tomato") +
  coord_flip() +
  labs(title = "Top 20 Studies by Sample Count",
       x = "Study Name",
       y = "Samples Count") +
  theme_minimal()

File: src/relative_abundance/analysis/plot_sample_count.py
import pandas as pd
import matplotlib.pyplot as plt

sample_csv = "./root:Host-associated:Human:Digestive%20system:Large%20intestine:Fecal_first.csv"

### Option 1: Plot Only the Top N Studies
# Read the CSV file into a DataFrame
df = pd.read_csv(sample_csv)

# Rename columns for convenience
df = df.rename(columns={
    "attributes.samples-count": "samples_count",
    "id": "study_id"
})

# Convert samples_count to numeric (if needed)
df["samples_count"] = pd.to_numeric(df["samples_count"], errors='coerce')

# Sort the DataFrame by sample count in descending order and take the top 20
top20 = df.sort_values("samples_count", ascending=False).head(20)

# Plot a horizontal bar graph for better readability
plt.figure(figsize=(10, 8))
plt.barh(top20["study_id"], top20["samples_count"], color='skyblue')
plt.xlabel("Samples Count")
plt.title("Top 20 Studies by Sample Count")
plt.gca().invert_yaxis()  # Highest value on top
plt.tight_layout()
plt.savefig("Top_20_sample_size.svg")

### Option 2: Plot a Summary or Aggregated View
# Read the CSV file
df = pd.read_csv(sample_csv)

df = df.rename(columns={
    "attributes.samples-count": "samples_count",
    "id": "study_id"
})
df["samples_count"] = pd.to_numeric(df["samples_count"], errors='coerce')

# Create bins for sample counts
bins = [0, 100, 300, 500, 1000, df["samples_count"].max()]
labels = ["0-100", "101-300", "301-500", "501-1000", "1001+"]

df['count_bin'] = pd.cut(df['samples_count'], bins=bins, labels=labels, include_lowest=True)

# Plot a bar chart of the bins
bin_counts = df['count_bin'].value_counts().sort_index()

plt.figure(figsize=(8, 6))
plt.bar(bin_counts.index.astype(str), bin_counts.values, color='salmon')
plt.xlabel("Samples Count Range")
plt.ylabel("Number of Studies")
plt.title("Distribution of Studies by Samples Count")
plt.tight_layout()
plt.savefig("Aggregate_sample_size.svg")

### Option 3: Plot All 320 Entries (Vertical Bars)
df = pd.read_csv(sample_csv)
df = df.rename(columns={
    "attributes.samples-count": "samples_count",
    "id": "study_id"
})
df["samples_count"] = pd.to_numeric(df["samples_count"], errors='coerce')

plt.figure(figsize=(10, 20))
plt.barh(df["study_id"], df["samples_count"], color='mediumseagreen')
plt.xlabel("Samples Count")
plt.title("Samples Count for 320 Studies")
plt.tight_layout()
plt.savefig("all_sample_size_count.svg")
